src/Widgets/DisplaysOverlay.vala

```diff
@@ -214,6 +214,7 @@ public class Display.DisplaysOverlay : Gtk.Overlay {
         display_widget.show_all ();
         display_widget.set_as_primary.connect (() => set_as_primary (display_widget.virtual_monitor));
         display_widget.check_position.connect (() => check_intersects (display_widget));
+        display_widget.move_display.connect ((diff_x, diff_y, event) => move_display (display_widget, diff_x, diff_y, event));
```

Comment:
You should not need to pass display_widget to the handler - the source of the event is automatically passed. If you change the handler to 
```
private void move_display (Gtk.Widget source, double diff_x, double diff_y) {
    var event = Gtk.get_current_event ();
    var display_widget = (DisplayWidget)source;
    ....
}
```
You can connect with 
```
display_widget.move_display.connect (move_display);
```
Or you could just write the whole handler as a closure.